### PR TITLE
Add Wikipedia info box to the company result page

### LINKED ISSUE #25: Add Wikipedia Info Box

### PR DESCRIPTION
**Pull request by alaacs**
Closes #25

### CODE REVIEW

File: nodeapp/app/view/showCompany/showCompany.js [hunk 1]
```diff
@@ -30,4 +31,56 @@ angular.module('csrLookupApp.showCompany', ['ngRoute'])
...
+   function fillWikipediaBox(companyWikipediaName)
+   {
+       var wikiData;
+        $.ajax({
+            type: "GET",
+            url: "http://en.wikipedia.org/w/api.php?action=parse&format=json&prop=text&section=0&callback=?&page="+companyWikipediaName,
+            contentType: "application/json; charset=utf-8",
+            async: false,
+            dataType: "json",
+            success: function (data, textStatus, jqXHR) {
```
Comment (lbraun) [inline nodeapp/app/view/showCompany/showCompany.js]: What does jqXHR mean here?

Comment (alaacs) [reply nodeapp/app/view/showCompany/showCompany.js]: Jquery XML HTTP Request 💃

Comment (lbraun) [reply nodeapp/app/view/showCompany/showCompany.js]: 👍 😃

File: app/app.css [hunk 2]
```diff
@@ -45,6 +45,7 @@
 
 .search-result-wrapper{
   width: 100%;
+  margin-bottom: 2%;
```
Comment (lbraun) [inline app/app.css]: Can we do without this if we use bootstrap styles?

File: app/index.html [hunk 3]
```diff
@@ -49,11 +49,14 @@
       </div>
     </nav>
 
-    <div ng-view style="margin-top: 5%;"></div>
+    <div class="container">
+      <div ng-view style="margin-top: 2%;"></div>
```
Comment (lbraun) [inline app/index.html]: Do we still need a margin-top here? Inline styling is ugly ☹️

File: app/index.html [hunk 4]
```diff
@@ -49,11 +49,14 @@
...
+    </div>
 
     <!-- Optional JavaScript -->
     <!-- jQuery first, then Popper.js, then Bootstrap JS -->
-    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="http://code.jquery.com/jquery-1.10.2.js"></script>
+    <!-- <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrit1y="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script> -->
```
Comment (lbraun) [inline app/index.html]: You can just remove commented out code like this to keep the codebase cleaner

File: app/view/showCompany/showCompany.html [hunk 5]
```diff
@@ -1,68 +1,66 @@
...
-            TODO: Fill with wikipedia info box
-          </p>
-          <a href='https://en.wikipedia.org/wiki/{{company.wikipedia_name}}' target='_blank'>
-            {{company.name}}
-          </a>
-        </div>
+  <div class="col-4 company-result-wikipedia-column">
+    <div class="card bg-light">
+      <div class="card-body">
+        <!-- <h4 class="card-title">Wikipedia info box</h4> -->
```
Comment (lbraun) [inline app/view/showCompany/showCompany.html]: Remove commented out code

File: app/view/showCompany/showCompany.js [hunk 6]
```diff
@@ -30,4 +31,59 @@ angular.module('csrLookupApp.showCompany', ['ngRoute'])
...
+                }
+                else 
+                {
+                    $("#wikipediaBox").html("This wikipedia page does not exist!");
+                }
+            },
+            error: function (errorMessage) {
+            }
+        });
+       // var wikiData;
```
Comment (lbraun) [inline app/view/showCompany/showCompany.js]: Remove commented out code

File: app/view/showCompany/showCompany.js [hunk 7]
```diff
@@ -30,4 +31,59 @@ angular.module('csrLookupApp.showCompany', ['ngRoute'])
...
+            success: function (data, textStatus, jqXHR) {
+                console.log(data);
+                wikiData = data;
+                $("#wikipediaBox").html('');
+                if(wikiData.parse)
+                {
+                    var html = wikiData.parse.text['*'];
+                    var parsedHtml = $.parseHTML( html );
+                    var infobox = $(parsedHtml[0]).find('table.infobox.vcard');
+                    //infobox[0].attributes['style'] = ''
```
Comment (lbraun) [inline app/view/showCompany/showCompany.js]: Remove commented out code

File: app/view/showCompany/showCompany.html [hunk 8]
```diff
@@ -1,68 +1,66 @@
...
+        <p class="card-text wikipedia-box" id = 'wikipediaBox'>
+          TODO: Fill with wikipedia info box
```
Comment (lbraun) [inline app/view/showCompany/showCompany.html]: Maybe we can replace with something like "Loading..." now?

File: app/view/showCompany/showCompany.js [hunk 9]
```diff
@@ -30,4 +31,59 @@ angular.module('csrLookupApp.showCompany', ['ngRoute'])
...
+                    //removing the width style from the result wikipedia infobox;
```
Comment (lbraun) [inline app/view/showCompany/showCompany.js]: Styling this is a pain, huh? How does wikipedia do it? We can make the column wider if it helps...

File: app/view/showCompany/showCompany.js [hunk 10]
```diff
@@ -30,4 +31,59 @@ angular.module('csrLookupApp.showCompany', ['ngRoute'])
   }, function errorCallback(response) {
     $scope.evidence_records = null
   });
+  
+   function fillWikipediaBox(companyWikipediaName)
+   {
+       var wikiData;
+        $.ajax({
```
Comment (lbraun) [inline app/view/showCompany/showCompany.js]: I think mostly we use 2 spaces for indentation. Could you update this to match that convention?

File: app/view/showCompany/showCompany.js (hunk 9 above)
Comment (alaacs) [reply app/view/showCompany/showCompany.js]: yeah it's pain, I moved the style to our style sheet but gave it !important to override the wikipedia inline style

File: app/index.html (hunk 4 above)
Comment (alaacs) [reply app/index.html]: I removed the unneeded comments, created a lib folder and downloaded all the used libraries locally.

File: app/index.html (hunk 3 above)
Comment (alaacs) [reply app/index.html]: yeah we need it, but moved it to the app.css

File: app/app.css (hunk 2 above)
Comment (alaacs) [reply app/app.css]: this is for the search results repeater to have space between search results items.

File: app/index.html (hunk 4 above)
Comment (lbraun) [reply app/index.html]: Wow this looks better. Do you know from your experience, is it more common to do it the first way or to download the libraries locally like this? I can see pros and cons to both.

**Comment by lbraun** [comment]
Thanks for the updates @alaacs. Looks good-to-go to me 👍 ✅

**Pull request merged**